jinahub/segmenters/VADSpeechSegmenter/vad_speech_segmenter.py

```diff
@@ -0,0 +1,135 @@
+    :param dump: a bool to specify whether to dump the segmented audio
+    :param repo: default repo name of silero-vad
+    :param model: default model name of silero-vad
+
+    """
+
+    TARGET_SAMPLE_RATE = 16000
+
+    def __init__(
+        self,
+        normalize: bool = False,
+        dump: bool = False,
+        repo: str = 'snakers4/silero-vad',
+        model: str = 'silero_vad',
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        if model is None or repo is None:
+            raise ValueError('model and repo cannot be None')
+
+        # TODO: remove the following temporary fix for torch hub
+        torch.hub._validate_not_a_forked_repo = lambda *args: True
+        self.model, utils = torch.hub.load(
+            repo_or_dir=repo, model=model, force_reload=True
+        )
+
+        (_, self.get_speech_ts_adaptive, self.save_audio, *_) = utils
+        self.normalize = self._normalize if normalize else lambda audio, _: audio
+        self.dump = dump
+
+    @requests
+    def segment(
+        self,
+        docs: Optional['DocumentArray'] = None,
+        parameters: Optional[Dict] = {},
+        **kwargs,
+    ):
+        """
+        Segment all audio docs to chunks using silero-vad
+        :param docs: documents sent to the segmenter. The docs must have either
+        `blob` or `uri` that ends with either `.mp3` or `.wav`.
+        :param parameters: dictionary to override the default parameters.
+        """
+
+        for doc in docs:
+            unnormalized_audio, sample_rate = self._resample(*self._load_raw_audio(doc))
+            audio = self.normalize(unnormalized_audio, sample_rate)
+            speech_timestamps = self.get_speech_ts_adaptive(audio, self.model)
+            doc.blob, doc.tags['sample_rate'] = (
+                audio.detach().cpu().numpy(),
+                sample_rate,
+            )
+            dump_dir = self.dump and (Path(self.workspace) / 'audio')
+
+            if dump_dir:
+                dump_dir.mkdir(exist_ok=True)
+                self.save_audio(
+                    str(dump_dir / f'doc_{doc.id}_original.wav'),
+                    unnormalized_audio,
+                    int(sample_rate),
+                )
+
+            chunk_path = f'{dump_dir}/chunk_{doc.id}_{{0}}_{{1}}.wav'
+            for ts in speech_timestamps:
+                doc.chunks.append(
+                    Document(
+                        blob=audio[ts['start'] : ts['end']].detach().cpu().numpy(),
+                        location=[ts['start'], ts['end']],
+                        tags=doc.tags,
+                    )
+                )
+
+                if dump_dir:
+                    self.save_audio(
+                        chunk_path.format(ts['start'], ts['end']),
+                        unnormalized_audio[ts['start'] : ts['end']],
+                        int(sample_rate),
+                    )
+
+    def _load_raw_audio(self, doc):
```

Comment:
Some type annotations would be nice (on all of these auxiliarry functions)

Reply:
Addressed, please see [here](https://github.com/jina-ai/executors/pull/97/commits/26ae02fc9d1b12c0c381f0495f4f1ae1d900b677) for the change.